src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -84,6 +94,34 @@ Signature LazyCreateSignature()
         private RuntimeType ReflectedTypeInternal => m_reflectedTypeCache.GetRuntimeType();
 
         internal BindingFlags BindingFlags => m_bindingFlags;
+
+
+        [DebuggerStepThrough]
+        [DebuggerHidden]
+        internal unsafe object InvokeNonEmitUnsafe(object? obj, IntPtr* args, Span<object?> argsForTemporaryMonoSupport, BindingFlags invokeAttr)
+        {
+            if ((invokeAttr & BindingFlags.DoNotWrapExceptions) == 0)
+            {
+                bool rethrow = false;
+
+                try
+                {
+                    return RuntimeMethodHandle.InvokeMethod(obj, (void**)args, Signature, isConstructor: obj is null, out rethrow)!;
+                }
+                catch (OutOfMemoryException)
```

Comment:
The special handling of OutOfMemory is also one of those special-cases that we can cleanup while are on it.

Reply:
Sure. It looks like it doesn't affect any tests. I'll push the new one-liner commit once this CI run is done.